docs/design/fabric-v2+/attestation-interface.md

```diff
@@ -0,0 +1,63 @@
+# Attestation Interface
+
+This document defines the interface for producing and verifying an attestation.
+
+
+## Producing an attestation
+
+This interface is only available inside an enclave.
+
+```
+int init_attestation(
+    uint8_t* params,
```

Comment:
What did you have in mind with these params?  Is this how you would pass spid and corresponding credentials to the library?

Reply:
> What did you have in mind with these params? Is this how you would pass spid and corresponding credentials to the library?

I would expect this to be a json blob (or similar). So it would contain spid, and maybe sigrl and api key if necessary.

> given that these params are originally on the untrusted side and also are needed there given the current epid sgx apis, probably this function is better done in fpc_stub rather than (fpc_stub_)enclave

They are on the untrusted side, which means either at (i) the peer or at (ii) the admin.

In case (i):
* these parameters should be defined in the peer configuration (core.yaml), unless we use admin calls that terminate in the Go chaincode. If, for any reason, the parameters should be changed, the peer should be reconfigured (maybe restarting it, or through admin calls).
* putting the init in untrusted side means that the `get_attestation` would not work without calling first the init in an abstraction below. For this reason, it does not seem appropriate to separate the two functions.

In case (ii):
* the parameters are kept by the admin, and can be provided through the `createenclave` which terminates inside the enclave. Three benefits here: 1) we already have the `createenclave` in place; 2) we can remove configuration parameters from the core.yaml; 3) we keep init and get_attestation side by side, making the api self-contained.

Reply:
> According to interfaces.md -- which as mentioned yesterday has not everything reflected yet in UML diagrams --, the createenclave ends in ecc which then calls ecc_enclave via init_enclave. So we would have to add the attestation parameter also there.

I would leave as it is now. Reason is: too many changes going on; once these stabilize, we can make them uniform.

Reply:
hmm, `interfaces.md` should be stable now and should have been the reference regarding these ecc/ecc-enclave interfaces? So i don't see when would be a better time to change it than now?